arch/sim/src/sim/up_internal.h

```diff
@@ -125,7 +125,8 @@ extern char **g_argv;
 
 /* Context switching */
 
-#if defined(CONFIG_HOST_X86_64) && !defined(CONFIG_SIM_M32)
+#if (defined(CONFIG_HOST_X86_64) && !defined(CONFIG_SIM_M32)) || \
+     defined(CONFIG_HOST_ARM64)
 void up_copyfullstate(unsigned long *dest, unsigned long *src);
```

Comment:
could handle all case by:
void up_copyfullstate(unsigned long *dest, unsigned long *src);

Reply:
done.

Reply:
Not working on Linux(sim-02):

    In file included from sim/up_unblocktask.c:35:
    sim/up_unblocktask.c: In function 'up_unblock_task':
    Error: sim/up_unblocktask.c:92:33: error: passing argument 1 of 'up_copyfullstate' from incompatible pointer type [-Werror=incompatible-pointer-types]
       92 |           up_savestate(rtcb->xcp.regs);
          |                        ~~~~~~~~~^~~~~
          |                                 |
          |                                 xcpt_reg_t * {aka unsigned int *}
    sim/up_internal.h:66:45: note: in definition of macro 'up_savestate'
       66 | #define up_savestate(regs) up_copyfullstate(regs, (xcpt_reg_t *)CURRENT_REGS)
          |                                             ^~~~
    sim/up_internal.h:128:38: note: expected 'long unsigned int *' but argument is of type 'xcpt_reg_t *' {aka 'unsigned int *'}
      128 | void up_copyfullstate(unsigned long *dest, unsigned long *src);
          |                       ~~~~~~~~~~~~~~~^~~~

Reply:
ditto